src/acl/RegexData.cc

```diff
@@ -197,19 +173,18 @@ compileOptimisedREs(std::list<RegexPattern> &curlist, const SBufList &sl, const
 
         if (reSize > 1024) { // must be < BUFSIZ everything included
             debugs(28, 2, "buffer full, generating new optimised RE..." );
-            if (!compileRE(newlist, accumulatedRE, flags))
-                return 0;
+            compileREs(newlist, accumulatedRE, flags);
             accumulatedRE.clear();
             reSize = 0;
             continue;    /* do the loop again to add the RE to largeRE */
         }
     }
 
-    if (!compileRE(newlist, accumulatedRE, flags))
-        return 0;
-
-    accumulatedRE.clear();
-    reSize = 0;
+    if (!accumulatedRE.empty()) {
+        compileREs(newlist, accumulatedRE, flags);
+        accumulatedRE.clear();
+        reSize = 0;
+    }
```

Comment:
After the cleanup in this PR, this code duplication became even more obvious. It is possible to refactor the code further to avoid the code duplication surrounding compileREs() calls, but that duplication existed before this PR, and such refactoring would require a lot of changes unrelated to this PR, so it is best done in a dedicated PR. We are not making things worse here.

Reply:
Question is *which* code duplication. I recall some of it being intentional, though not that exact reasons. If you take on the de-dupe work be careful.